File: MATLAB/estimators/est_GaussLSQ1_2D.m
function xest=est_GaussLSQ1_2D(photonsi,patternpos,L,sigma,iscenter)
    pi=photonsi/sum(photonsi);
    Ls2=L^2/8/sigma^2;
    % eq 2.63
    if iscenter
        numrim=size(patternpos,1)-iscenter;
        xest=(numrim+exp(Ls2))/Ls2/numrim*sum(pi.*patternpos);
    else
        xest=sum(pi.*patternpos)/Ls2;
    end
end

File: python/estimators/est_GaussLSQ1_2D.py
import numpy as np

def est_GaussLSQ1_2D(photonsi, patternpos, L, sigma, iscenter, *args):
    pi = photonsi / np.sum(photonsi)
    Ls2 = L**2 / 8 / sigma**2
    if len(args) > 0 and iscenter:
        numrim = patternpos.shape[0] - 1
        xest = (numrim + np.exp(Ls2)) / Ls2 / numrim * np.sum(pi * patternpos, axis=0)
    else:
        xest = np.sum(pi*patternpos, axis=0) / Ls2
        
    return xest
